### PR TITLE
Clean up certbot-auto docs

### PR DESCRIPTION
**Pull request by bmw**
Fixes https://github.com/certbot/certbot/issues/8519.

I left the `certbot-auto` docs in `install.rst` to avoid breaking links and to help propagate information about our changes there. I moved it closer to the bottom of the doc though since I think our documentation about OS packages and Docker is more helpful to most people.

Screenshots of the docs where any significant changes are made are below and the "our snaps" link I added works.

### compatibility.rst:
Screen Shot 2020-12-14 at 9 53 52 AM
### install.rst:
Screen Shot 2020-12-14 at 10 06 11 AM

### CODE REVIEW

File: certbot/docs/install.rst [hunk 1]
```diff
@@ -303,6 +235,33 @@ OS packaging is an ongoing effort. If you'd like to package
...
+------------
+
+We used to have a shell script named ``certbot-auto`` to help people install
+Certbot on UNIX operating systems, however, this script is no longer supported.
+
+Problems with Python virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using ``certbot-auto`` on a low memory system such as VPS with less than
+512MB of RAM, the required dependencies of Certbot may fail to build.  This can
```
Comment (alexzorin) [inline certbot/docs/install.rst:250]: What is the reason to keep this section around? 

Are there other (non-development) circumstances in which it is relevant, like the `pipx` proposal perhaps?

If so, shall we reword it to not mention `certbot-auto`?

Comment (bmw) [reply certbot/docs/install.rst:250]: My motivation for keeping this around was indeed to not break the anchor, especially when `certbot-auto` isn't quite fully deprecated yet and it links to it. See https://github.com/certbot/certbot/blob/5151e2afee98adebce061215ecda6a739c4f2ecd/certbot-auto#L1730

It currently does not affect any of our non-development installation instructions. This problem can occur when compiling cryptography/cffi and in snaps, Docker, and OS packages it is precompiled. (I suppose a user of a source based distribution like Gentoo could hit this, although not in a Python virtual environment as described in the title of this section, but I've never seen this in practice.)

This could affect whatever becomes of our `pipx` proposal, but I think this is largely a resolved problem. This used to be a very common issue (see https://github.com/certbot/certbot/issues/1081) but since then `cryptography` and `cffi` have shipped precompiled wheels for amd64 and aarch64. We could maybe keep this around in anticipation of the `pipx` instructions in case anyone hits this issue, but I personally think this is rare enough that it adds more noise than it's worth. I'm interested in your opinion, but mine is we should make this certbot-auto specific and delete it at some point, but if we see (m)any users confused by this in the future, we could always add the section back or generalize it.

Comment (alexzorin) [reply certbot/docs/install.rst:250]: Ah, OK.

I was under the impression that there were no longer any systems that would call `pip install` anymore, unless they purposefully evaded the deprecation by cloning from an earlier tag or something.

If the code still runs on some end-user systems, I think it's fine to keep it as-is.

Otherwise, I think that fully removing it is appropriate, as it has no end-user audience.

Comment (bmw) [reply certbot/docs/install.rst:250]: certbot-auto will be fully deprecated in our next release, but since we haven't released that change yet, I have a slight preference to keep this around so we don't have to worry about accidentally breaking the anchor on the website before our next release.

**Review by alexzorin — CHANGES_REQUESTED** [review]
There's a couple of paras under *System Requirements* that look like they can go:

>Additional integrity verification of certbot-auto script can be done by verifying its digital signature. This requires a local installation of gpg2, which comes packaged in many Linux distributions under name gnupg or gnupg2.

If we're not providing instructions anywhere on how to verify the script, then it doesn't serve any purpose to describe this.

>
>Installing with certbot-auto requires 512MB of RAM in order to build some of the dependencies. Installing from pre-built OS packages avoids this requirement. You can also temporarily set a swap file. See “Problems with Python virtual environment” below for details.

Up to you. I say that the anchor and error message in the certbot-auto section is sufficient, but I don't mind.

**Comment by bmw** [comment]
Thanks for the review and catching that. I missed those references to certbot-auto when working on this.

I removed those paragraphs and a screenshot of the relevant section now looks like:
Screen Shot 2020-12-16 at 9 20 58 AM

File: certbot/docs/install.rst [hunk 2]
```diff
@@ -78,74 +67,6 @@ choosing "snapd" in the "System" dropdown menu. (You should select "snapd"
...
-The ``certbot-auto`` command updates to the latest client release automatically.
-Since ``certbot-auto`` is a wrapper to ``certbot``, it accepts exactly
-the same command line flags and arguments. For more information, see
-`Certbot command-line options <https://certbot.eff.org/docs/using.html#command-line-options>`_.
-
-For full command line help, you can type::
-
-  /usr/local/bin/certbot-auto --help all
-
-Problems with Python virtual environment
```
Comment (adferrand) [inline certbot/docs/install.rst:132]: Related to https://github.com/certbot/certbot/pull/8391, I wanted to note that the workaround to handle low memory system will certainly be relevant for an installation procedure that would use pip. We are planning to add this with https://github.com/certbot/certbot/issues/8422.

It may be relevant to create an issue or add it to the documentation in https://github.com/certbot/certbot/issues/8422 to remember about this, including the proposition to use `dd` instead of `fallocate` as said in https://github.com/certbot/certbot/pull/8391.

Comment (bmw) [reply certbot/docs/install.rst:132]: Did you read my comment at https://github.com/certbot/certbot/pull/8532#discussion_r542948604? I argued that this is largely a historical problem, suggested we not worry about it, and we can revisit that decision if (m)any people hit it which I'm not confident they will.

Comment (adferrand) [reply certbot/docs/install.rst:132]: Ah sorry. Indeed I read it yesterday and forgot about it today while studying the other PR about this section.

So indeed it is not really useful anymore.

**Review by alexzorin — APPROVED** [review]
